pandas/core/frame.py

```diff
@@ -2641,7 +2641,7 @@ def reindexer(value):
             from pandas.core.series import _sanitize_index
 
             # turn me into an ndarray
-            value = _sanitize_index(value, self.index, copy=False)
+            value = _sanitize_index(value, self.index, copy=True)
```

Comment:
I am not convinced that you *always* need to copy. I think only in very certain circumstances should you copy. It depends if this array is *already* copied previously or not. Put some tests in place then it will be clear.